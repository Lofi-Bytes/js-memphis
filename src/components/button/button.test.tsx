import React from 'react'
import renderer from 'react-test-renderer'

import { axe, toHaveNoViolations } from 'jest-axe'
import { render } from '@testing-library/react'

import Button from './button'


expect.extend(toHaveNoViolations)

describe('Button', () => {
  /**
   * Snapshot
   * Primary
   * Secondary
   * Tertiary
   */
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Button
          action="secondary"
          className="w-full"
          disabled={false}
          role="link"
          title="Learn more about me"
          to="/about"
        >
          Learn more about me&nbsp;&nbsp;<i className="fa-regular fa-chevron-right" />
        </Button>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Button
          action="secondary"
          className="w-full"
          disabled={false}
          role="link"
          title="Learn more about me"
          to="/about"
        >
          Learn more about me&nbsp;&nbsp;<i className="fa-regular fa-chevron-right" />
        </Button>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Button
          action="tertiary"
          disabled={false}
          role="link"
          title="Return to homepage"
          className="mt-6 w-max"
          to="/"
        >
          <i className="fas fa-arrow-left" />&nbsp;&nbsp;Return to homepage
        </Button>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  /**
   * Accessibility
   * Primary
   * Secondary
   * Tertiary
   */
  it('should have no violations', async () => {
    const { container } = render(
      <Button
        action="primary"
        className="mt-4"
        disabled={false}
        role="button"
        title="Submit"
        type="submit"
      >
        Submit
      </Button>
    )
    const results = await axe(container)

    expect(results).toHaveNoViolations()
  })

  it('should have no violations', async () => {
    const { container } = render(
      <Button
        action="secondary"
        className="w-full"
        disabled={false}
        role="link"
        title="Learn more about me"
        to="/about"
      >
        Learn more about me&nbsp;&nbsp;<i className="fa-regular fa-chevron-right" />
      </Button>
    )
    const results = await axe(container)

    expect(results).toHaveNoViolations()
  })

  it('should have no violations', async () => {
    const { container } = render(
      <Button
        action="tertiary"
        disabled={false}
        role="link"
        title="Return to homepage"
        className="mt-6 w-max"
        to="/"
      >
        <i className="fas fa-arrow-left" />&nbsp;&nbsp;Return to homepage
      </Button>
    )
    const results = await axe(container)

    expect(results).toHaveNoViolations()
  })
})
