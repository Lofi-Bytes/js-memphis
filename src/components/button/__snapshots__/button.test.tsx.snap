// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<a
  className="block cursor-pointer font-medium outline-none focus:outline-none hover:outline-none active:outline-none overflow-hidden text-center tracking-wide transition active:shadow-sm duration-200 ring-1 ring-offset-1 focus:ring-1 focus:ring-offset-1 hover:ring-1 hover:ring-offset-1 active:ring-1 active:ring-offset-1 px-6 py-2 rounded-full shadow-lg shadow-purple-300/50 transform hover:-translate-y-0.5 focus:-translate-y-0.5 active:translate-y-0.5 ease-in-out active:ring-offset-purple-700 active:ring-purple-200 active:ring-opacity-75 hover:ring-offset-purple-700 hover:ring-purple-200 hover:ring-opacity-75 focus:ring-offset-purple-700 focus:ring-purple-200 focus:ring-opacity-75 ring-offset-purple-300 ring-purple-200 ring-opacity-75 bg-purple-50 text-purple-600 w-full"
  href="/about"
  role="link"
  title="Learn more about me"
>
  Learn more about me  
  <i
    className="far fa-chevron-right"
  />
</a>
`;

exports[`Button renders correctly 2`] = `
<a
  className="block cursor-pointer font-medium outline-none focus:outline-none hover:outline-none active:outline-none overflow-hidden text-center tracking-wide transition active:shadow-sm duration-200 ring-1 ring-offset-1 focus:ring-1 focus:ring-offset-1 hover:ring-1 hover:ring-offset-1 active:ring-1 active:ring-offset-1 px-6 py-2 rounded-full shadow-lg shadow-purple-300/50 transform hover:-translate-y-0.5 focus:-translate-y-0.5 active:translate-y-0.5 ease-in-out active:ring-offset-purple-700 active:ring-purple-200 active:ring-opacity-75 hover:ring-offset-purple-700 hover:ring-purple-200 hover:ring-opacity-75 focus:ring-offset-purple-700 focus:ring-purple-200 focus:ring-opacity-75 ring-offset-purple-300 ring-purple-200 ring-opacity-75 bg-purple-50 text-purple-600 w-full"
  href="/about"
  role="link"
  title="Learn more about me"
>
  Learn more about me  
  <i
    className="far fa-chevron-right"
  />
</a>
`;

exports[`Button renders correctly 3`] = `
<a
  className="block cursor-pointer font-medium outline-none focus:outline-none hover:outline-none active:outline-none overflow-hidden text-center tracking-wide transition border-b-2 border-transparent ease-in-out text-purple-700 transform hover:border-purple-700 focus:border-purple-700 active:border-purple-700 mt-6 w-max"
  href="/"
  role="link"
  title="Return to homepage"
>
  <i
    className="fas fa-arrow-left"
  />
    Return to homepage
</a>
`;
